src/advanced-settings/AdvancedSettings.jsx

```diff
@@ -35,19 +54,30 @@ const AdvancedSettings = ({ intl, courseId }) => {
     }
     setEditedSettings((prevEditedSettings) => ({
       ...prevEditedSettings,
-      [settingName]: value || ' ',
+      [settingName]: removeExtraQuotes(value) || ' ',
     }));
   };
 
   const handleResetSettingsValues = () => {
+    showErrorModal(false);
     setEditedSettings({});
     showSaveSettingsPrompt(false);
   };
 
   const handleUpdateAdvancedSettingsData = () => {
-    dispatch(updateCourseAppSetting(courseId, parseArrayOrObjectValues(editedSettings)));
-    showSaveSettingsPrompt(false);
-    window.scrollTo({ top: 0, behavior: 'smooth' });
+    validateAdvancedSettingsData(editedSettings, setErrorFields)
+        .then(() => {
```

Comment:
nit: something wrong with indents? should be 2 spaces I think

Reply:
indent: 'off' :)